### ISSUE TITLE
SolarTimes: SolarNoon should have the same date as the requested date

### ISSUE THREAD

**Issue by mfeingol** [porrey/Solar-Calculator#1]
Instead, it's set to the current date.

This can be worked around, but ideally it would be the same date as sunrise, sunset, aka the requested date.

Here's what I'm seeing in VS with 2.0.1:

```
-		solar	{Innovative.SolarCalculator.SolarTimes}	Innovative.SolarCalculator.SolarTimes
+		AtmosphericRefraction	{0°.8330}	Innovative.Geometry.Angle
		EccentricityOfEarthOrbit	0.0167011631360033348253412462	decimal
		EquationOfTime	11.874755288182159680353068853	decimal
+		ForDate	{10/6/2017 12:00:00 AM -07:00}	System.DateTimeOffset
+		HourAngleSunrise	{87°.08060873}	Innovative.Geometry.Angle
		JulianCentury	0.1776260552133242071640428939	decimal
		JulianDay	2458032.7916666666666666666667	decimal
+		Latitude	{37°.17404886}	Innovative.Geometry.Angle
+		Longitude	{-113°.01283544}	Innovative.Geometry.Angle
+		MeanEclipticObliquity	{23°.43698123}	Innovative.Geometry.Angle
+		ObliquityCorrection	{23°.43497811}	Innovative.Geometry.Angle
+		SolarDeclination	{-5°.21791966}	Innovative.Geometry.Angle
+		SolarNoon	{10/1/2017 12:20:10 PM}	System.DateTime
+		SunApparentLongitude	{193°.21880195}	Innovative.Geometry.Angle
+		SunEquationOfCenter	{-1°.9137307}	Innovative.Geometry.Angle
+		SunGeometricMeanLongitude	{195°.14119911}	Innovative.Geometry.Angle
+		SunMeanAnomaly	{6751°.89839959}	Innovative.Geometry.Angle
+		SunTrueLongitude	{193°.22746842}	Innovative.Geometry.Angle
+		SunlightDuration	{11:36:38.6920000}	System.TimeSpan
+		Sunrise	{10/6/2017 6:31:51 AM}	System.DateTime
+		Sunset	{10/6/2017 6:08:29 PM}	System.DateTime
		TimePastLocalMidnight	0	decimal
		TimeZoneOffset	-7	decimal
		TrueSolarTime	1419.8234135131501596803530688	decimal
		VarY	0.0430182430890316477981212096	decimal
```

**Comment by porrey** [comment]
The _Solar Noon_ calculation, which is a time calculation, was being added to **DateTime.Now.Date** instead of the **this.ForDate.Date**. This has been corrected and released in v2.0.2. Unit test cases were added to check the output dates of **Sunrise**, **Sunset** and **SolarNoon**. I have verified this in an independent test.

Please verify and let me know if your issue is resolved and I will close the issue.

**Comment by mfeingol** [comment]
Fixed in 2.0.2, thank you!
